### ISSUE TITLE
Fetching package info on non-pypa link 

### ISSUE THREAD

**Issue by Abhishek-Dev09-zz** [jazzband/pip-tools#1179]
#### Environment Versions

1. OS Type
1. Python version: `3.6.8`
1. pip version:  `20.1.1`
1. pip-tools version: `5.2.1`

#### Steps to replicate

`pip-compile --find-links=thirdparty --generate-hashes --index-url=https://github.com/nexB/scancode-toolkit/tree/develop/thirdparty`

#### Actual result
`Fetch package info from PyPI failed: https://github.com/nexB/scancode-toolkit/tree/develop/pypi/soupsieve/json: HTTPSConnectionPool(host='github.com', port=443): Max retries exceeded with url: /nexB/scancode-toolkit/tree/develop/pypi/soupsieve/json (Caused by ResponseError('too many 500 error responses',))
    Couldn't get hashes from PyPI, fallback to hashing files
    Hashing soupsieve-1.9.2-py2.py3-none-any.whl
  pygments`

#### Expected result - it would directly generate hash from file. For example-
`Hashing soupsieve-1.9.2-py2.py3-none-any.whl
  pygments`

**Comment by atugushev** [comment]
Hello @Abhishek-Dev09,

Thanks for the issue! That should work:

```console
$ git clone https://github.com/nexB/scancode-toolkit
$ pip-compile --generate-hashes --find-links=scancode-toolkit/thirdparty 
```

**Comment by atugushev** [comment]
You cannot pass GitHub tree URL to [--index-url](https://pip.pypa.io/en/stable/reference/pip_install/#cmdoption-index-url), see the doc:

> Base URL of the Python Package Index (default https://pypi.org/simple). This should point to a repository compliant with PEP 503 (the simple repository API) or a local directory laid out in the same format.

**Comment by Abhishek-Dev09** [comment]
> Hello @Abhishek-Dev09,
> 
> Thanks for the issue! That should work:
> 
> ```
> $ git clone https://github.com/nexB/scancode-toolkit
> $ pip-compile --generate-hashes --find-links=scancode-toolkit/thirdparty 
> ```

@atugushev :  I do not want hashing from pypi instead it fetches from `find-links` options  because deps of my requirenment.txt is not available on pypi.

**Comment by atugushev** [comment]
@Abhishek-Dev09 

Ah, I see your point. A package, let's say, `six` from `--find-links=./links/` gets its hashes from PyPI, but it should get hashes from a file `./links/six-1.10.tar.gz.` in links directory. Thanks!

**Comment by Abhishek-Dev09** [comment]
> @Abhishek-Dev09
> 
> Ah, I see your point. A package, let's say, `six` from `--find-links=./links/` gets its hashes from PyPI, but it should get hashes from a file `./links/six-1.10.tar.gz.` in links directory. Thanks!

@atugushev : Yes, you understand my point.

**Comment by Abhishek-Dev09** [comment]
@pombredanne As we are dependent on this tools , soon this bug will be solved , then it get hash directly from wheel .

**Comment by atugushev** [comment]
Hashing packages from find links was fixed in #1723.
